{
  "url": "https://api.github.com/repos/facebook/react/issues/4575",
  "repository_url": "https://api.github.com/repos/facebook/react",
  "labels_url": "https://api.github.com/repos/facebook/react/issues/4575/labels{/name}",
  "comments_url": "https://api.github.com/repos/facebook/react/issues/4575/comments",
  "events_url": "https://api.github.com/repos/facebook/react/issues/4575/events",
  "html_url": "https://github.com/facebook/react/issues/4575",
  "id": 99403995,
  "node_id": "MDU6SXNzdWU5OTQwMzk5NQ==",
  "number": 4575,
  "title": "React.createElement(null, null) - should it not throw?",
  "user": {
    "login": "mik01aj",
    "id": 3027415,
    "node_id": "MDQ6VXNlcjMwMjc0MTU=",
    "avatar_url": "https://avatars3.githubusercontent.com/u/3027415?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/mik01aj",
    "html_url": "https://github.com/mik01aj",
    "followers_url": "https://api.github.com/users/mik01aj/followers",
    "following_url": "https://api.github.com/users/mik01aj/following{/other_user}",
    "gists_url": "https://api.github.com/users/mik01aj/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/mik01aj/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/mik01aj/subscriptions",
    "organizations_url": "https://api.github.com/users/mik01aj/orgs",
    "repos_url": "https://api.github.com/users/mik01aj/repos",
    "events_url": "https://api.github.com/users/mik01aj/events{/privacy}",
    "received_events_url": "https://api.github.com/users/mik01aj/received_events",
    "type": "User",
    "site_admin": false
  },
  "labels": [

  ],
  "state": "closed",
  "locked": false,
  "assignee": null,
  "assignees": [

  ],
  "milestone": null,
  "comments": 2,
  "created_at": "2015-08-06T10:20:12Z",
  "updated_at": "2015-08-07T06:52:36Z",
  "closed_at": "2015-08-06T23:10:12Z",
  "author_association": "NONE",
  "body": "When I do:\n\n```\nReact.createElement(undefined, null)\n```\n\n(this sometimes happens because of a bug in my application)\n\nIt just gives a warning:\n\n```\nWarning: React.createElement: type should not be null or undefined. It should be a string (for DOM elements) or a ReactClass (for composite components).\n```\n\nAnd only when rendering, it fails with one more warning and an error:\n\n```\nWarning: Only functions or strings can be mounted as React components.\nreact.js:9259\nUncaught TypeError: Cannot read property 'toUpperCase' of undefined\n    at autoGenerateWrapperClass (https://facebook.github.io/react/js/react.js:9259:18)\n    at Object.getComponentClassForElement (https://facebook.github.io/react/js/react.js:13201:49)\n    at ReactCompositeComponentMixin._processProps (https://facebook.github.io/react/js/react.js:6615:44)\n    at ReactCompositeComponentMixin.mountComponent (https://facebook.github.io/react/js/react.js:6313:28)\n    at wrapper [as mountComponent] (https://facebook.github.io/react/js/react.js:13427:21)\n    at Object.ReactReconciler.mountComponent (https://facebook.github.io/react/js/react.js:14128:35)\n    at Object.__REACT_INSPECTOR_RUNTIME__0_13_1.React.ReactReconciler.mountComponent (<anonymous>:106:41)\n    at mountComponentIntoNode (https://facebook.github.io/react/js/react.js:12038:32)\n    at ReactReconcileTransaction.Mixin.perform (https://facebook.github.io/react/js/react.js:16683:20)\n    at batchedMountComponentIntoNode (https://facebook.github.io/react/js/react.js:12059:15)\n```\n\nThis makes bugs harder to debug, because in some cases (e.g. server-side rendering) the warnings can get lost in a log and only the error is visible (saying 'Uncaught TypeError: Cannot read property 'toUpperCase' of undefined'). Why are these warnings not errors then? I suppose that when element type is null, there is not much use of such an element and it's better to throw a meaningful error early.\n\nAlso, if the error is thrown earlier, I will get the stacktrace of the function call that actually passed `null` to the constructor. Then I can debug it easier.\n\nI like the fact that you give warnings about stuff, but I think that in some cases errors would be even more helpful.\n",
  "closed_by": {
    "login": "sophiebits",
    "id": 6820,
    "node_id": "MDQ6VXNlcjY4MjA=",
    "avatar_url": "https://avatars2.githubusercontent.com/u/6820?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/sophiebits",
    "html_url": "https://github.com/sophiebits",
    "followers_url": "https://api.github.com/users/sophiebits/followers",
    "following_url": "https://api.github.com/users/sophiebits/following{/other_user}",
    "gists_url": "https://api.github.com/users/sophiebits/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/sophiebits/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/sophiebits/subscriptions",
    "organizations_url": "https://api.github.com/users/sophiebits/orgs",
    "repos_url": "https://api.github.com/users/sophiebits/repos",
    "events_url": "https://api.github.com/users/sophiebits/events{/privacy}",
    "received_events_url": "https://api.github.com/users/sophiebits/received_events",
    "type": "User",
    "site_admin": false
  }
}
